default/python/AI/AIDependencies.py

```diff
@@ -668,14 +668,14 @@ class Tags:
 
 # <editor-fold desc="Piloting traits">
 # TODO (Morlic): Consider using only 1 dict with tuple for (Capacity, SecondaryStat) effects
-PILOT_DAMAGE_MODIFIER_DICT = {
+PILOT_DAMAGE_MODIFIER_DICT = {trait: tuple({weapon_name: damage * SHIP_WEAPON_DAMAGE_FACTOR for weapon_name, damage in part_damage.items()}.items()) for trait, part_damage in {
```

Comment:
1) Line too long
2) expression is super compex
3) variable type was changed and usage was not changed.  This is a critical blocker!


Here is my version of the refactoring with the wrong type. 
```
from typing import Dict, Tuple

SHIP_WEAPON_DAMAGE_FACTOR = 1.0  # TEMP VARIABLE


_TRAIT_MODIFIERS = {
    # TRAIT:    {weapon_name: effect, weapon_name2: effect2,...}
    "NO":       {},
    "BAD":      {"SR_WEAPON_1_1": -1, "SR_WEAPON_2_1": -2, "SR_WEAPON_3_1": -3, "SR_WEAPON_4_1": -5},
    "GOOD":     {"SR_WEAPON_1_1":  1, "SR_WEAPON_2_1":  2, "SR_WEAPON_3_1":  3, "SR_WEAPON_4_1": 5},
    "GREAT":    {"SR_WEAPON_1_1":  2, "SR_WEAPON_2_1":  4, "SR_WEAPON_3_1":  6, "SR_WEAPON_4_1": 10},
    "ULTIMATE": {"SR_WEAPON_1_1":  3, "SR_WEAPON_2_1":  6, "SR_WEAPON_3_1":  9, "SR_WEAPON_4_1": 15, "SR_WEAPON_0_1": 1
 }
}



def _get_part_damage_with_factor(part_damage: Dict[str, int]) -> Tuple[Tuple[str, float]]:
    updated_items = ((weapon_name, damage * SHIP_WEAPON_DAMAGE_FACTOR) for weapon_name, damage in part_damage.items())
    return tuple(updated_items)


PILOT_DAMAGE_MODIFIER_DICT = {
    trait: _get_part_damage_with_factor(part_damage) for trait, part_damage in _TRAIT_MODIFIERS.items()
}
```

Reply:
I split the comprehension for better comprehension and fixed the types. I let some AI play for 200 turns and no related errors in logs or message window.